controller/api/destination/server.go

```diff
@@ -193,30 +225,30 @@ func getHostAndPort(dest *pb.GetDestination) (string, int, error) {
 		var err error
 		port, err = strconv.Atoi(hostPort[1])
 		if err != nil {
-			err = fmt.Errorf("Invalid port %s", hostPort[1])
-			log.Error(err)
-			return "", 0, err
+			return "", 0, fmt.Errorf("Invalid port %s", hostPort[1])
 		}
 	}
-	return host, port, nil
+	return host, watcher.Port(port), nil
 }
 
-func buildResolver(
-	k8sDNSZone string,
-	k8sAPI *k8s.API,
-) (streamingDestinationResolver, error) {
-	k8sDNSZoneLabels := []string{}
-	if k8sDNSZone != "" {
-		var err error
-		k8sDNSZoneLabels, err = splitDNSName(k8sDNSZone)
-		if err != nil {
-			return nil, err
-		}
+func (s *server) getServiceAndPort(dest *pb.GetDestination) (watcher.ServiceID, watcher.Port, error) {
+	if dest.Scheme != "k8s" {
```

Comment:
Let's please not perpetuate this logic. I'd suggest that we just ignore scheme checks until we have a reason the check them.